db/model/sample.js

```diff
@@ -170,32 +170,44 @@ module.exports = function sample(seq, dataTypes) {
        *  encountered while performing the sample upsert operation itself.
        */
       upsertByName(toUpsert, isBulk) {
-        let subjasp;
+        let sampleExists = true;
         return new seq.Promise((resolve, reject) => {
-          u.getSubjectAndAspectBySampleName(seq, toUpsert.name)
-          .then((sa) => {
-            subjasp = sa;
-            toUpsert.subjectId = sa.subject.id;
-            toUpsert.aspectId = sa.aspect.id;
-          })
-          .then(() => Sample.findOne({
+          // get sample by name
+          Sample.findOne({
             where: {
-              subjectId: subjasp.subject.id,
-              aspectId: subjasp.aspect.id,
-              isDeleted: NO,
+              name: {
+                $iLike: toUpsert.name,
+              },
```

Comment:
@shriramshankar After recent changes, we are indexing only name, so we are using iLike to match case-insensitive name.
And I am not able to find an operator to match case-insensitive without regex. So, using iLike only.